### PR TITLE
[Merged by Bors] - feat (FieldTheory/Adjoin): generalize `IntermediateField.exists_algHom_of_splits`

### PR DESCRIPTION
**Pull request by alreadydone**
+ Extract from `Lifts.exists_lift_of_splits` a version that assumes integrality and splitting of minimal polynomial over an intermediate field (either actual IntermediateField or middle of IsScalarTower) rather than the base field

+ Generalize `exists_algHom_adjoin_of_splits`, `exists_algHom_of_adjoin_splits`, `exists_algHom_of_splits` to the same setting

+ Prove the [extension lemma](https://math.stackexchange.com/a/687914/12932) `surjective_comp_algebraMap_of_isAlgebraic/isSeparable` to be used in the proof that an epimorphism in the category of fields is a purely inseparable extension.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/FieldTheory/Adjoin.lean [hunk 1]
```diff
@@ -1244,26 +1244,72 @@ theorem Lifts.exists_upper_bound (c : Set (Lifts F E K)) (hc : IsChain (· ≤ 
...
 /-- Given an element `s : E` whose conjugates are all in `K`, any lift can be extended to one
   whose carrier contains `s`. -/
-theorem Lifts.exists_lift_of_splits (x : Lifts F E K) {s : E} (h1 : IsIntegral F s)
-    (h2 : (minpoly F s).Splits (algebraMap F K)) : ∃ y, x ≤ y ∧ s ∈ y.carrier :=
-  have I1 : IsIntegral x.carrier s := h1.tower_top
-  have I2 := (minpoly.degree_pos I1).ne'
-  have key : (minpoly x.carrier s).Splits x.emb.toRingHom :=
-    splits_of_splits_of_dvd _ (map_ne_zero (minpoly.ne_zero h1))
-      ((splits_map_iff _ _).2 (x.emb.comp_algebraMap ▸ h2)) (minpoly.dvd_map_of_isScalarTower _ _ _)
+theorem Lifts.exists_lift_of_splits' (x : Lifts F E K) {s : E} (h1 : IsIntegral x.carrier s)
```
Comment (acmepjz) [inline Mathlib/FieldTheory/Adjoin.lean:1247]: Perhaps the docstrings should be changed? The old `Lifts.exists_lift_of_splits` is "whose conjugates over `F` are all in `K`", and the new one is "whose conjugates over `x.carrier` are all in `K`".

**Review by acmepjz — COMMENTED** [review]
Looks good to me.

File: Mathlib/FieldTheory/Adjoin.lean (hunk 1 above)
Comment (alreadydone) [reply Mathlib/FieldTheory/Adjoin.lean]: Thanks! This was an oversight. I shouldn't have modified the copy of the theorem that carries the docstring with it ...

File: Mathlib/FieldTheory/IsAlgClosed/Basic.lean [hunk 2]
```diff
@@ -209,8 +209,16 @@ instance (priority := 100) IsAlgClosure.separable (R K : Type*) [Field R] [Field
 
 namespace IsAlgClosed
 
-variable (K : Type u) [Field K] (L : Type v) (M : Type w) [Field L] [Algebra K L] [Field M]
-  [Algebra K M] [IsAlgClosed M] (hL : Algebra.IsAlgebraic K L)
+variable {K : Type u} [Field K] {L : Type v} {M : Type w} [Field L] [Algebra K L] [Field M]
+  [Algebra K M] [IsAlgClosed M]
+
+theorem surjective_comp_algebraMap_of_isAlgebraic {E : Type*}
```
Comment (riccardobrasca) [inline Mathlib/FieldTheory/IsAlgClosed/Basic.lean:215]: Can you add a docstring to this one (and to the other version)? Maybe also a reference, for example is this Lemma 4.13 of http://alpha.math.uga.edu/~pete/FieldTheory.pdf ?

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

**Pull request closed without merging**